pkg/api/validation/validation.go

```diff
@@ -687,11 +687,10 @@ func ValidatePodStatusUpdate(newPod, oldPod *api.Pod) errs.ValidationErrorList {
 
 	allErrs = append(allErrs, ValidateObjectMetaUpdate(&oldPod.ObjectMeta, &newPod.ObjectMeta).Prefix("metadata")...)
 
-	// TODO: allow change when bindings are properly decoupled from pods
```

Comment:
By default pod status does not contain host(IP). It used to be filled in pod_cache. Also, if we don't allow to change this, why not just overwrite it to the last known value instead of returning error?